### ISSUE TITLE
Building a tree of bodies

### ISSUE THREAD

**Issue by DomWilliams0** [sebcrozet/nphysics#110]
I'm trying to procedurally generate a tree of bodies that are each fixed to their parent with a fixed joint, but `World.step` is panicking with `Index out of bounds.`

In this example program, I'm using `add_multibody_link` to build a small binary tree with 7 nodes. Could anyone please point me in the right direction?

```rust
extern crate nalgebra;
extern crate ncollide3d;
extern crate nphysics3d;

use nalgebra::{zero, Isometry3, Vector3};
use ncollide3d::shape::*;
use nphysics3d::joint::*;
use nphysics3d::object::*;
use nphysics3d::volumetric::*;
use nphysics3d::world::World;

type N = f64;

fn new_shape<J: Joint<N>>(world: &mut World<N>, joint: J, parent: BodyHandle) -> BodyHandle {
    let shape = ShapeHandle::new(Cuboid::new(Vector3::new(1.0, 1.0, 1.0)));
    world.add_multibody_link(
        parent,
        joint,
        zero(),
        zero(),
        shape.inertia(1.0),
        shape.center_of_mass(),
    )
}
fn main() {
    let mut world: World<N> = World::new();

    let root = new_shape(
        &mut world,
        FreeJoint::new(Isometry3::identity()),
        BodyHandle::ground(),
    );

    // two children under root
    let a = new_shape(&mut world, FixedJoint::new(Isometry3::identity()), root);
    let b = new_shape(&mut world, FixedJoint::new(Isometry3::identity()), root);

    // two children each to a and b
    new_shape(&mut world, FixedJoint::new(Isometry3::identity()), a);
    new_shape(&mut world, FixedJoint::new(Isometry3::identity()), a);

    new_shape(&mut world, FixedJoint::new(Isometry3::identity()), b);
    new_shape(&mut world, FixedJoint::new(Isometry3::identity()), b);

    world.step(); // panics
}
```
```
thread 'main' panicked at 'Index out of bounds.', /home/dom/.cargo/registry/src/github.com-1ecc6299db9ec823/nphysics3d-0.8.1/src/utils/index_mut2.rs:24:9
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
             at libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::sys_common::backtrace::print
             at libstd/sys_common/backtrace.rs:71
             at libstd/sys_common/backtrace.rs:59
   2: std::panicking::default_hook::{{closure}}
             at libstd/panicking.rs:207
   3: std::panicking::default_hook
             at libstd/panicking.rs:223
   4: std::panicking::rust_panic_with_hook
             at libstd/panicking.rs:402
   5: std::panicking::begin_panic
             at /checkout/src/libstd/panicking.rs:365
   6: <alloc::vec::Vec<T> as nphysics3d::utils::index_mut2::IndexMut2<usize>>::index_mut2
             at ./<panic macros>:3
   7: nphysics3d::utils::index_mut2::IndexMut2::index_mut_const
             at /home/dom/.cargo/registry/src/github.com-1ecc6299db9ec823/nphysics3d-0.8.1/src/utils/index_mut2.rs:15
   8: <nphysics3d::object::multibody::Multibody<N>>::update_inertias
             at /home/dom/.cargo/registry/src/github.com-1ecc6299db9ec823/nphysics3d-0.8.1/src/object/multibody.rs:706
   9: <nphysics3d::object::multibody::Multibody<N>>::update_dynamics
             at /home/dom/.cargo/registry/src/github.com-1ecc6299db9ec823/nphysics3d-0.8.1/src/object/multibody.rs:518
  10: <nphysics3d::object::body_set::BodySet<N>>::update_dynamics
             at /home/dom/.cargo/registry/src/github.com-1ecc6299db9ec823/nphysics3d-0.8.1/src/object/body_set.rs:132
  11: <nphysics3d::world::world::World<N>>::step
             at /home/dom/.cargo/registry/src/github.com-1ecc6299db9ec823/nphysics3d-0.8.1/src/world/world.rs:215
  12: testy::main
             at src/main.rs:46
  13: std::rt::lang_start::{{closure}}
             at /checkout/src/libstd/rt.rs:74
  14: std::panicking::try::do_call
             at libstd/rt.rs:59
             at libstd/panicking.rs:306
  15: __rust_maybe_catch_panic
             at libpanic_unwind/lib.rs:102
  16: std::rt::lang_start_internal
             at libstd/panicking.rs:285
             at libstd/panic.rs:361
             at libstd/rt.rs:58
  17: std::rt::lang_start
             at /checkout/src/libstd/rt.rs:74
  18: main
  19: __libc_start_main
  20: _start

```

**Comment by sebcrozet** [comment]
That’s a bug. I will try to fix this tomorrow evening!

By the way, asuming you add colliders as well, keep in mind that using a rigid body with a `Compound` shape (initialised as your tree) will result in a faster simulation than using a multibody with only fixed joints between links. But I guess that might not be possible for you since this require you to generate le full tree before adding it to the physics world.

**Comment by sebcrozet** [comment]
@DomWilliams0 Fixed in v0.8.3!

**Comment by DomWilliams0** [comment]
Fantastic, thank you very much!
